wls-briefwahl-service/src/test/java/de/muenchen/oss/wahllokalsystem/briefwahlservice/service/beanstandetewahlbriefe/BeanstandeteWahlbriefeModelMapperTest.java

```diff
@@ -0,0 +1,100 @@
+import de.muenchen.oss.wahllokalsystem.wls.common.security.domain.BezirkIDUndWaehlerverzeichnisNummer;
+import java.util.HashMap;
+import lombok.val;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mapstruct.factory.Mappers;
+
+class BeanstandeteWahlbriefeModelMapperTest {
+
+    private final BeanstandeteWahlbriefeModelMapper unitUnderTest = Mappers.getMapper(BeanstandeteWahlbriefeModelMapper.class);
+
+    @Nested
+    class ToEmbeddedId {
+
+        @Test
+        void nullInNullOut() {
+            Assertions.assertThat(unitUnderTest.toEmbeddedId(null)).isNull();
+        }
+
+        @Test
+        void objectIsMapped() {
+            val wahlbezirkID = "wbzId";
+            val waehlerverzeichnisNummer = 123L;
+            val reference = new BeanstandeteWahlbriefeReference(wahlbezirkID, waehlerverzeichnisNummer);
+
+            val expectedResult = new BezirkIDUndWaehlerverzeichnisNummer(wahlbezirkID, waehlerverzeichnisNummer);
+
+            val result = unitUnderTest.toEmbeddedId(reference);
+
+            Assertions.assertThat(result).isEqualTo(expectedResult);
+        }
+    }
+
+    @Nested
+    class ToModel {
+
+        @Test
+        void nullInNullOut() {
+            Assertions.assertThat(unitUnderTest.toModel(null)).isNull();
+        }
+
+        @Test
+        void objectIsMapped() {
+            val wahlbezirkID = "wbzId";
+            val waehlerverzeichnisNummer = 123L;
+            val id = new BezirkIDUndWaehlerverzeichnisNummer(wahlbezirkID, waehlerverzeichnisNummer);
+            val zurueckweisungen = new HashMap<String, Zurueckweisungsgrund[]>();
+            zurueckweisungen.put("wahl1", new Zurueckweisungsgrund[] { Zurueckweisungsgrund.ZUGELASSEN, Zurueckweisungsgrund.NICHT_WAHLBERECHTIGT });
+            zurueckweisungen.put("wahl2", new Zurueckweisungsgrund[] { Zurueckweisungsgrund.ZUGELASSEN, Zurueckweisungsgrund.NICHT_WAHLBERECHTIGT,
+                    Zurueckweisungsgrund.UNTERSCHRIFT_FEHLT });
+            val beanstandeteWahlbriefe = new BeanstandeteWahlbriefe(id, zurueckweisungen);
+
+            val expectedZurueckweisungen = new HashMap<String, Zurueckweisungsgrund[]>();
+            expectedZurueckweisungen.put("wahl1", new Zurueckweisungsgrund[] { Zurueckweisungsgrund.ZUGELASSEN, Zurueckweisungsgrund.NICHT_WAHLBERECHTIGT });
+            expectedZurueckweisungen.put("wahl2", new Zurueckweisungsgrund[] { Zurueckweisungsgrund.ZUGELASSEN, Zurueckweisungsgrund.NICHT_WAHLBERECHTIGT,
+                    Zurueckweisungsgrund.UNTERSCHRIFT_FEHLT });
+            val expecetedResult = new BeanstandeteWahlbriefeModel(wahlbezirkID, waehlerverzeichnisNummer, expectedZurueckweisungen);
+
+            val result = unitUnderTest.toModel(beanstandeteWahlbriefe);
+
+            Assertions.assertThat(result).usingRecursiveComparison().isEqualTo(expecetedResult);
+        }
+    }
+
+    @Nested
+    class ToEntity {
```

Comment:
Könnte man noch auf den return bzw. true hit testen.
Sieht gut aus 👍  finde sonst nichts mehr

Reply:
Wie meinst du das?

Reply:
![grafik](https://github.com/it-at-m/Wahllokalsystem/assets/162574031/2374acbd-54b1-4665-9554-69825270c6fe)

Diese Methoden im MapperImpl

Reply:
Da die Klasse generiert ist sehe ich nicht unbedingt die Notwendigkeit einer hohen Code-Coverage. Ich schlage vor wir besprechen es am Montag im Team.

Reply:
Da vertraue ich dir wenn du das sagst 😄